doc/src/extras.rst

```diff
@@ -141,6 +141,262 @@ Logging cursor
 
 .. autoclass:: MinTimeLoggingCursor
 
+Replication cursor
+^^^^^^^^^^^^^^^^^^
+
+.. autoclass:: ReplicationConnection
+
+   This connection factory class can be used to open a special type of
+   connection that is used for streaming replication.
+
+   Example::
+
+       from psycopg2.extras import ReplicationConnection, REPLICATION_PHYSICAL, REPLICATION_LOGICAL
+       conn = psycopg2.connect(dsn, connection_factory=ReplicationConnection)
```

Comment:
Does the user need to supply `replication=database` or `replication=on` in the `dsn`? Or do you add that?

If connecting to 9.3 or older, which don't support `replication=database` and only do "physical" replication, how's that handled?

(I'm intentionally reading docs before code so I see from a user perspective and know what the docs don't cover).